File: Python/Problems/Problems/problem1619.py
from typing import List, Dict

from .problem import Problem


class Problem1619(Problem):
    def test(self) -> bool:
        arr = [1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 3]

        expected = 2.0

        result = self.trimMean(arr)

        print(result)

        return result == expected

    def trimMean(self, arr: List[int]) -> float:
        arr.sort()
        n = len(arr)
        outliners = n // 20

        sum = 0
        for i in range(outliners, n - outliners):
            sum += arr[i]

        return sum / (n - 2 * outliners)

File: C++/Problems/Problems/problem_1619.cpp
#include "problem_1619.h"

#include <algorithm>

bool problem_1619::test() {
    std::vector arr { 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 3 };

    auto expected = 2.0;

    auto result = trimMean(arr);

    std::cout << result << std::endl;

    return result == expected;
}

double problem_1619::trimMean(std::vector<int> &arr) {
    std::ranges::sort(arr);
    int n = arr.size();
    int outliners = n / 20;

    int sum = 0;
    for (int i = outliners; i < n - outliners; i++)
    {
        sum += arr[i];
    }

    return (double)sum / (n - 2 * outliners);
}
